### ISSUE TITLE
Gnome-shell crashed and dumped core when the extension turns on

### ISSUE THREAD

**Issue by z-engine** [OttoAllmendinger/gnome-shell-imgur#45]
Bugged extension now, leads to gnome-shell crashed and dumped core. 
I test it on two PC, with Archlinux. Problem the same. I could't login to the system after install gnome-shell-imgur. Gnome-shell crashed every time at the time of login and throws me back in gdm.
In journalctl logs i saw many errors and stack traces of gnome-shell threads. 

```
Apr 03 16:33:18 zen systemd-coredump[4375]: Process 4333 (gnome-shell) of user 1000 dumped core                                       
    Stack trace of thread 4333:
    #0  0x00007f3357539a10 raise (libc.so.6)
    #1  0x00007f335753b13a abort (libc.so.6)
    #2  0x00007f3357b39345 g_assertion_message (libglib-2.0.so.0)
    #3  0x00007f3357b393da g_assertion_message_expr (libglib-2.0.so.0)
    #4  0x00007f3358b251f3 gtk_icon_info_load_icon_finish (libgtk-3.so.0)
    #5  0x00007f3358b25278 n/a (libgtk-3.so.0)
    #6  0x00007f33580b7303 n/a (libgio-2.0.so.0)
    #7  0x00007f33580b7339 n/a (libgio-2.0.so.0)
    #8  0x00007f3357b1245a g_main_context_dispatch (libglib-2.0.so.0)
    #9  0x00007f3357b12810 n/a (libglib-2.0.so.0)
    #10 0x00007f3357b12b32 g_main_loop_run (libglib-2.0.so.0)
    #11 0x00007f33592c2aac meta_run (libmutter.so.0)
    #12 0x0000000000402197 main (gnome-shell)
    #13 0x00007f3357526511 __libc_start_main (libc.so.6)
    #14 0x00000000004022ca n/a (gnome-shell)
    
    Stack trace of thread 4334:
    #0  0x00007f33575e867d poll (libc.so.6)
    #1  0x00007f3357b127a6 n/a (libglib-2.0.so.0)
    #2  0x00007f3357b128bc g_main_context_iteration (libglib-2.0.so.0)
    #3  0x00007f3357b12901 n/a (libglib-2.0.so.0)
    #4  0x00007f3357b3a175 n/a (libglib-2.0.so.0)
    #5  0x00007f33578b12e7 start_thread (libpthread.so.0)
    #6  0x00007f33575f254f __clone (libc.so.6)
    
    Stack trace of thread 4350:
    #0  0x00007f33575e867d poll (libc.so.6)
    #1  0x00007f3353940ee1 n/a (libpulse.so.0)
    #2  0x00007f33539326f1 pa_mainloop_poll (libpulse.so.0)
    #3  0x00007f3353932d8e pa_mainloop_iterate (libpulse.so.0)
    #4  0x00007f3353932e40 pa_mainloop_run (libpulse.so.0)
    #5  0x00007f3353940e29 n/a (libpulse.so.0)
    #6  0x00007f3349218fe8 n/a (libpulsecommon-10.0.so)
    #7  0x00007f33578b12e7 start_thread (libpthread.so.0)
    #8  0x00007f33575f254f __clone (libc.so.6)
    
    Stack trace of thread 4349:
    #0  0x00007f33575e867d poll (libc.so.6)
    #1  0x00007f3357b127a6 n/a (libglib-2.0.so.0)
    #2  0x00007f3357b128bc g_main_context_iteration (libglib-2.0.so.0)
    #3  0x00007f333cdc54bd n/a (libdconfsettings.so)
    #4  0x00007f3357b3a175 n/a (libglib-2.0.so.0)
    #5  0x00007f33578b12e7 start_thread (libpthread.so.0)
    #6  0x00007f33575f254f __clone (libc.so.6)
    
    Stack trace of thread 4369:
    #0  0x00007f33575ed889 syscall (libc.so.6)
    #1  0x00007f3357b580ca g_cond_wait_until (libglib-2.0.so.0)
    #2  0x00007f3357ae6e89 n/a (libglib-2.0.so.0)
    #3  0x00007f3357b3ab46 n/a (libglib-2.0.so.0)
    #4  0x00007f3357b3a175 n/a (libglib-2.0.so.0)
    #5  0x00007f33578b12e7 start_thread (libpthread.so.0)
    #6  0x00007f33575f254f __clone (libc.so.6)
    
    Stack trace of thread 4351:
    #0  0x00007f33578b7756 pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
    #1  0x00007f334a9624e0 PR_WaitCondVar (libnspr4.so)
    #2  0x00007f3353396c0e n/a (libmozjs-24.so)
    #3  0x00007f334a967d6c n/a (libnspr4.so)
    #4  0x00007f33578b12e7 start_thread (libpthread.so.0)
    #5  0x00007f33575f254f __clone (libc.so.6)
    
    Stack trace of thread 4352:
    #0  0x00007f33578b7756 pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
    #1  0x00007f334a9624e0 PR_WaitCondVar (libnspr4.so)
    #2  0x00007f335340abab n/a (libmozjs-24.so)
    #3  0x00007f334a967d6c n/a (libnspr4.so)
    #4  0x00007f33578b12e7 start_thread (libpthread.so.0)
    #5  0x00007f33575f254f __clone (libc.so.6)
    
    Stack trace of thread 4337:
    #0  0x00007f33575e867d poll (libc.so.6)
    #1  0x00007f3357b127a6 n/a (libglib-2.0.so.0)
    #2  0x00007f3357b12b32 g_main_loop_run (libglib-2.0.so.0)
    #3  0x00007f33580f8446 n/a (libgio-2.0.so.0)
    #4  0x00007f3357b3a175 n/a (libglib-2.0.so.0)
    #5  0x00007f33578b12e7 start_thread (libpthread.so.0)
    #6  0x00007f33575f254f __clone (libc.so.6)
    
    Stack trace of thread 4348:
    #0  0x00007f33575e867d poll (libc.so.6)
    #1  0x00007f334d4a0f0f n/a (libresolv.so.2)
    #2  0x00007f334d49eeb4 __libc_res_nquery (libresolv.so.2)
    #3  0x00007f334d49f525 n/a (libresolv.so.2)
    #4  0x00007f334d49f9d9 __libc_res_nsearch (libresolv.so.2)
    #5  0x00007f32ffcfc9f3 _nss_dns_gethostbyname4_r (libnss_dns.so.2)
    #6  0x00007f3304c2fefb _nss_resolve_gethostbyname4_r (libnss_resolve.so.2)
    #7  0x00007f33575dcb04 gaih_inet.constprop.5 (libc.so.6)
    #8  0x00007f33575dde47 getaddrinfo (libc.so.6)
    #9  0x00007f33580ba6d0 n/a (libgio-2.0.so.0)
    #10 0x00007f33580b7add n/a (libgio-2.0.so.0)
    #11 0x00007f3357b3ab6e n/a (libglib-2.0.so.0)
    #12 0x00007f3357b3a175 n/a (libglib-2.0.so.0)
    #13 0x00007f33578b12e7 start_thread (libpthread.so.0)
    #14 0x00007f33575f254f __clone (libc.so.6)
    
    Stack trace of thread 4336:
    #0  0x00007f33578b7ca6 pthread_cond_timedwait@@GLIBC_2.3.2 (libpthread.so.0)
    #1  0x00007f3345177e44 n/a (libGLX_nvidia.so.0)
    #2  0x00007f333efd5394 n/a (libnvidia-glcore.so.378.13)
    #3  0x00007f334517712c n/a (libGLX_nvidia.so.0)
    #4  0x00007f33578b12e7 start_thread (libpthread.so.0)
    #5  0x00007f33575f254f __clone (libc.so.6)
-- Subject: Process 4333 (gnome-shell) dumped core
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
-- Documentation: man:core(5)
-- 
-- Process 4333 (gnome-shell) crashed and dumped core.
-- 
-- This usually indicates a programming error in the crashing program and
-- should be reported to its vendor as a bug.

```

I just log to free tty and remove user dir with this extension and it all worked. I try to install via browser and from git repo, the behavior is the same after the extension is turned on. 
Maybe this information will be useful: exactly the same problem with the extension [Caffeine](https://github.com/eonpatapon/gnome-shell-extension-caffeine)
System: Arch linux, Linux zen 4.10.6-1-zen, GNOME Shell 3.22.3 on wayland, systemd v. 232-8

**Comment by OttoAllmendinger** [comment]
Ouch! Will try to reproduce.

Does it happen with https://github.com/OttoAllmendinger/gnome-shell-screenshot as well?

**Comment by OttoAllmendinger** [comment]
Couldn't reproduce with same config. I remember some instability the last time I tried linux-zen on Arch. Does it also happen with stock linux kernel?

**Comment by z-engine** [comment]
gnome-shell-screenshot works fine, without problem. I try to switch stock linux kernel and let you know about the results

**Comment by z-engine** [comment]
So, I tested on a standard kernel, but everything is the same. But i saw in log next line:
`Apr 04 11:31:05 zen gnome-shell[6812]: Cannot open pixbuf loader module file '/usr/lib/gdk-pixbuf-2.0/2.10.0/loaders.cache': No such file or directory`
Then I try to create `loaders.cache` with command 
`sudo sh -c 'gdk-pixbuf-query-loaders > /usr/lib/gdk-pixbuf-2.0/2.10.0/loaders.cache'`
and now this extension and Caffeine works fine, but Screenshot Location continues to crash.
I tried to reproduce it a few times, deleting loaders.cache and trying to install the extension, then again create it and install and it and this solution works, at least for me.
